drivers/accel/adxl355/iio_adxl355.c

```diff
@@ -0,0 +1,1117 @@
+static int adxl355_iio_read_samples(void* dev, int* buff, uint32_t samples);
+static int adxl355_iio_update_channels(void* dev, uint32_t mask);
+
+/******************************************************************************/
+/************************ Variable Declarations ******************************/
+/******************************************************************************/
+static struct iio_attribute adxl355_iio_temp_attrs[] = {
+	{
+		.name = "offset",
+		.show = adxl355_iio_read_offset,
+		.store = NULL,
+	},
+	{
+		.name = "raw",
+		.show = adxl355_iio_read_raw,
+		.store = NULL,
+	},
+	{
+		.name = "scale",
+		.show = adxl355_iio_read_scale,
+		.store = NULL,
+	},
+	{
+		.name = "converted",
+		.show = adxl355_iio_read_converted,
+		.store = NULL,
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+static struct iio_attribute adxl355_iio_accel_attrs[] = {
+	{
+		.name  = "calibbias",
+		.show  = adxl355_iio_read_calibbias,
+		.store = adxl355_iio_write_calibbias,
+	},
+	{
+		.name   = "filter_high_pass_3db_frequency",
+		.shared = IIO_SHARED_BY_TYPE,
+		.show   = adxl355_iio_read_hpf,
+		.store  = adxl355_iio_write_hpf,
+	},
+	{
+		.name   = "filter_high_pass_3db_frequency_available",
+		.shared = IIO_SHARED_BY_TYPE,
+		.show   = adxl355_iio_read_hpf_available,
+	},
+	{
+		.name = "raw",
+		.show = adxl355_iio_read_raw,
+	},
+	{
+		.name   = "sampling_frequency",
+		.shared = IIO_SHARED_BY_TYPE,
+		.show   = adxl355_iio_read_sampling_freq,
+		.store  = adxl355_iio_write_sampling_freq,
+	},
+	{
+		.name   = "sampling_frequency_available",
+		.shared = IIO_SHARED_BY_TYPE,
+		.show   = adxl355_iio_read_samp_freq_avail,
+	},
+	{
+		.name   = "scale",
+		.shared = IIO_SHARED_BY_TYPE,
+		.show   = adxl355_iio_read_scale,
+	},
+	{
+		.name = "converted",
+		.show = adxl355_iio_read_converted,
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+static struct scan_type adxl355_iio_accel_scan_type = {
+	.sign = 's',
+	.realbits = 20,
+	.storagebits = 32,
+	.shift = 0,
+	.is_big_endian = false
+};
+
+static struct scan_type adxl355_iio_temp_scan_type = {
+	.sign = 's',
+	.realbits = 12,
+	.storagebits = 32,
+	.shift = 0,
+	.is_big_endian = false
+};
+
+#define ADXL355_ACCEL_CHANNEL(index, reg, axis) { \
+	.ch_type = IIO_ACCEL,                         \
+	.channel = index,                             \
+	.address = reg,                               \
+	.modified = true,                             \
+	.channel2 = IIO_MOD_##axis,                   \
+	.scan_type = &adxl355_iio_accel_scan_type,    \
+	.scan_index = index,                          \
+	.attributes = adxl355_iio_accel_attrs,        \
+	.ch_out = false                               \
+}
+
+static struct iio_channel adxl355_channels[] = {
+	ADXL355_ACCEL_CHANNEL(0, chan_x, X),
+	ADXL355_ACCEL_CHANNEL(1, chan_y, Y),
+	ADXL355_ACCEL_CHANNEL(2, chan_z, Z),
+	{
+		.ch_type = IIO_TEMP,
+		.scan_type = &adxl355_iio_temp_scan_type,
+		.scan_index = 3,
+		.attributes = adxl355_iio_temp_attrs,
+		.ch_out = false,
```

Comment:
Is it worthwhile to initialize the `.channel` member here as well? By default it will be zero, coinciding with the X axis channel. Does it have to be the same or different?